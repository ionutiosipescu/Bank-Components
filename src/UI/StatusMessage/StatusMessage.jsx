import React from "react";
import { useEffect } from "react";
import { useState } from "react";
import { MessageContainer } from "./StatusMessage.style";
import { PropTypes } from "prop-types";

function StatusMessage({ ...props }) {
  const { type, text, size, handleShow, show } = props;

  const [showMessage, setShowMessage] = useState(true);
  const [isHiding, setIsHiding] = useState(false);

  useEffect(() => {
    const timer = setTimeout(() => {
      setIsHiding(true);
    }, 4500);

    return () => clearTimeout(timer);
  }, [show]);

  useEffect(() => {
    if (!isHiding) return;

    const timer = setTimeout(() => {
      setShowMessage(false);
      // handleShow();
    }, 300);

    return () => clearTimeout(timer);
  }, [isHiding]);

  return (
    showMessage && (
      <MessageContainer type={type} size={size} hide={isHiding}>
        <h3>{`${text}`.charAt(0).toUpperCase() + text.slice(1)}</h3>
      </MessageContainer>
    )
  );
}

StatusMessage.propTypes = {
  type: PropTypes.oneOf(["success", "error", "warning", "info", "default"]),
  text: PropTypes.string,
  size: PropTypes.oneOf(["fit", "half", "full", "xl"]),
};

export default StatusMessage;
