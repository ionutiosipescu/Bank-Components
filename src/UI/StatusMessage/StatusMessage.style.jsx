import styled, { css, keyframes } from "styled-components";
import { statusSize } from "../../../utils/componentSizes/statusMessageSizes";

const slideIn = keyframes`
  from {
    transform: translateX(-100%);
  }

  to {
    transform: translateX(0);
  }
`;

const shake = keyframes`
  0% {
    transform: translateX(0);
  }

  10% {
    transform: translateX(-10px);
  }

  20% {
    transform: translateX(10px);
  }

  30% {
    transform: translateX(-10px);
  }

  40% {
    transform: translateX(10px);
  }

  50% {
    transform: translateX(0);
  }
`;

const slideOut = keyframes`
  from {
    transform: translateX(0) scaleX(1);
  }

  to {
    transform: translateX(-50%) scaleX(0);
  }
`;

export const MessageContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: fit-content;
  position: relative;
  border-radius: 7px;
  padding: 15px 20px 15px 20px;
  margin: 5px;
  animation: ${slideIn} 0.2s ease-in-out forwards,
    ${shake} 0.3s ease-in-out forwards;
  ${(props) =>
    props.hide
      ? css`
          animation: ${slideOut} 0.3s ease-in-out forwards;
        `
      : ""}
  ${(props) =>
    props.size === "fit"
      ? statusSize.fit
      : props.size === "half"
      ? statusSize.half
      : props.size === "full"
      ? statusSize.full
      : statusSize.full}
  ${(props) =>
    props.type === "success"
      ? "background-color: var(--green-light); color: var(--green-dark);"
      : props.type === "error"
      ? "background-color: var(--red-light); color: var(--red-dark);"
      : props.type === "warning"
      ? "background-color: var(--yellow-light); color: var(--yellow-dark);"
      : props.type === "info"
      ? "background-color: var(--blue-light); color: var(--blue-dark);"
      : "background-color: var(--gray-light); color: var(--gray-dark);"}
`;
