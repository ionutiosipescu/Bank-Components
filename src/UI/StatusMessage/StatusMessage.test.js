import React from "react";
import { shallow } from "enzyme";
import StatusMessage from "./StatusMessage";
import { MessageContainer } from "./StatusMessage.style";

describe("StatusMessage component", () => {
  it("should render with correct props", () => {
    const props = {
      type: "success",
      text: "Test message",
      size: "full",
      handleShow: jest.fn(),
      show: true,
    };

    const wrapper = shallow(<StatusMessage {...props} />);

    expect(wrapper.find(MessageContainer).props().type).toEqual(props.type);
    expect(wrapper.find(MessageContainer).props().size).toEqual(props.size);
    expect(wrapper.find("h3").text()).toEqual("Test message");
    expect(wrapper).toMatchSnapshot();
  });
});
