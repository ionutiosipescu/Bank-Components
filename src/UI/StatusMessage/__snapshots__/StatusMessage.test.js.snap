// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMessage component should render with correct props 1`] = `
<styled.div
  hide={false}
  size="full"
  type="success"
>
  <h3>
    Test message
  </h3>
</styled.div>
`;
